src/Features/Core/Portable/AddConstructorParametersFromMembers/AddConstructorParametersFromMembersCodeRefactoringProvider.cs

```diff
@@ -151,5 +154,37 @@ static CodeAction GetOptionalContructorParametersCodeAction(Document document, C
                     document, constructorCandidate, containingType, missingOptionalParameters, useSubMenuName);
             }
         }
+
+        public async Task<ImmutableArray<IntentProcessorResult>> ComputeIntentAsync(Document priorDocument, TextSpan priorSelection, Document currentDocument, string serializedIntentData, CancellationToken cancellationToken)
+        {
+            var actions = await AddConstructorParametersFromMembersAsync(priorDocument, priorSelection, cancellationToken).ConfigureAwait(false);
+            if (actions.IsEmpty)
+            {
+                return ImmutableArray<IntentProcessorResult>.Empty;
+            }
+
+            // This provider generates nested actions, so we need to flatten them before we compute the edits.
+            actions = actions.SelectMany(a => GetNestedActions(a)).ToImmutableArray();
+
+            using var _ = ArrayBuilder<IntentProcessorResult>.GetInstance(out var results);
+
+            foreach (var action in actions)
+            {
+                var addParamAction = (AddConstructorParametersCodeAction)action;
```

Comment:
as per above, would far prefer if you just got the individual actions in this strongly typed form.   we can discuss offline if you want more thoughts here.